package Integracion.Querys;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

import Integracion.Excepciones.ExcepcionIntegracion;
import Integracion.Transacciones.GestorTransacciones;

public class ClientesProductosEnFechaDeterminada implements Query {

	@Override
	public Object ejecutar(Object parametro) throws ExcepcionIntegracion {
		
		try {
			
			GestorTransacciones.getInstancia().nuevaTransaccion();
			Connection conexion=(Connection) GestorTransacciones.getInstancia().getTransaccion().getResource();

			PreparedStatement ps = conexion.prepareStatement();
			
			ps.setInt(1, (int)parametro);
			ResultSet rs =ps.executeQuery();

			
			rs.close();
			
			return clientes;
			
		} catch (SQLException e) {
			
			
			throw new ExcepcionIntegracion("Error en la consulta");
		}
		
	}

}
